test/e2e/mount_test.go

```diff
@@ -81,80 +83,139 @@ func replaceTemplate(templateFile string, destFile string) error {
+		check(execCmd(exec.Command("kubectl", "apply", "-f", clusterFile)))
+		check(execCmd(exec.Command("kubectl", "wait", "containercluster/"+f.testClusterName,
+			"--for=condition=Ready", "--timeout", "15m")))
+
+		// Get kubeconfig to use to authenticate to test cluster
+		f.kubeconfigFile = filepath.Join(f.tempDir, "test-cluster-kubeconfig")
+		gcloudCmd := exec.Command("gcloud", "container", "clusters", "get-credentials", f.testClusterName,
+			"--zone", zone, "--project", f.testProjectID)
+		gcloudCmd.Env = append(os.Environ(), "KUBECONFIG="+f.kubeconfigFile)
+		check(execCmd(gcloudCmd))
+
+		// Install Secret Store
+		check(execCmd(exec.Command("kubectl", "apply", "--kubeconfig", f.kubeconfigFile,
+			"-f", fmt.Sprintf("https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/%s/deploy/rbac-secretproviderclass.yaml", f.secretStoreVersion),
+			"-f", fmt.Sprintf("https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/%s/deploy/rbac-secretprovidersyncing.yaml", f.secretStoreVersion),
+			"-f", fmt.Sprintf("https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/%s/deploy/csidriver.yaml", f.secretStoreVersion),
+			"-f", fmt.Sprintf("https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/%s/deploy/secrets-store.csi.x-k8s.io_secretproviderclasses.yaml", f.secretStoreVersion),
+			"-f", fmt.Sprintf("https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/%s/deploy/secrets-store.csi.x-k8s.io_secretproviderclasspodstatuses.yaml", f.secretStoreVersion),
+			"-f", fmt.Sprintf("https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/%s/deploy/secrets-store-csi-driver.yaml", f.secretStoreVersion),
+		)))
+
+		// Install GCP Plugin and Workload Identity bindings
+		check(execCmd(exec.Command("kubectl", "apply", "--kubeconfig", f.kubeconfigFile,
+			"-f", pluginFile)))
+
+		// Create test secret
+		secretFile := filepath.Join(f.tempDir, "secretValue")
+		check(ioutil.WriteFile(secretFile, []byte(f.testSecretID), 0644))
+		check(execCmd(exec.Command("gcloud", "secrets", "create", f.testSecretID, "--replication-policy", "automatic",
+			"--data-file", secretFile, "--project", f.testProjectID)))
+	} else {
+		type metadataStruct struct {
+			name string `yaml:"name"`
+		}
+
+		type audienceStruct struct {
+			audience string `yaml:"audience"`
+		}
+
+		type specStruct struct {
+			podInfoOnMount       bool             `yaml:"podInfoOnMount"`
+			attachRequired       bool             `yaml:"attachRequired"`
+			volumeLifecycleModes []string         `yaml:"volumeLifecycleModes"`
+			tokenRequests        []audienceStruct `yaml:"tokenRequests"`
+		}
+
+		type driver struct {
+			apiVersion string         `yaml:"apiVersion"`
+			kind       string         `yaml:"kind"`
+			metadata   metadataStruct `yaml:"metadata"`
+			spec       specStruct     `yaml:"spec"`
+		}
+
+		aud := audienceStruct{
+			audience: "secretmanager-csi-build.svc.id.goog",
```

Comment:
a comment here explaining the audience will probably help future devs realize this is specific to running the tests in the`secretmanager-csi-build` project.

Reply:
Comment added